pkg/kubelet/server/server.go

```diff
@@ -878,6 +891,69 @@ func (s *Server) getPortForward(request *restful.Request, response *restful.Resp
 	proxyStream(response.ResponseWriter, request.Request, url)
 }
 
+// checkpoint handles the checkpoint API request. It checks if the requested
+// podNamespace, pod and container actually exist and only then calls out
+// to the runtime to actually checkpoint the container.
+func (s *Server) checkpoint(request *restful.Request, response *restful.Response) {
+	pod, ok := s.host.GetPodByName(request.PathParameter("podNamespace"), request.PathParameter("podID"))
+	if !ok {
+		response.WriteError(http.StatusNotFound, fmt.Errorf("pod does not exist"))
+		return
+	}
+
+	containerName := request.PathParameter("containerName")
+
+	found := false
+	for _, container := range pod.Spec.Containers {
```

Comment:
this means we are unable to checkpoint init containers or ephemeral containers.

can you add a check for `pod.Spec.InitContainers` and a check for `pod.Spec.EphemeralContainers` if the feature gate for ephemeral containers is enabled?

Reply:
Thanks for catching this. I changed it and was able to checkpoint init and ephemeral containers.